### ISSUE TITLE
SequenceInArrayWithLargestSum a bit confusing

### ISSUE THREAD

**Issue by quinnbaetz** [dabaitu/300#10]
https://github.com/dabaitu/300/blob/master/math/sequenceInArrayWithLargestSum/python/solution.py

SequenceInArrayWithLargestSum was a bit convoluted and hard to follow.  I believe the code below solves it faster and is pretty easy to follow, can anyone prove me wrong?
```
#walks down array counting, if the counter ever falls below 0 it resets, otherwise it keeps adding up more!
def contiguous_seq(array):
	current_index = 0
	max_sum = -10000000000
	current_sum = 0
	for i, val in enumerate(array):
		current_sum += val;
		if current_sum > max_sum:
			max_sum = current_sum
		if current_sum < 0:
			current_sum = 0
	return max_sum


#if you need to factor in indices as well:
def contiguous_seq(array):
	current_index = 0
	#max_index = 0
	#max_length = 1
	max_sum = -10000000000
	current_sum = 0
	#floating_index = 0
	for i, val in enumerate(array):
		current_sum += val;
		if current_sum > max_sum:
			max_sum = current_sum
			#max_index = floating_index
			#max_length = i - max_index
		if current_sum < 0:
			current_sum = 0
			#floating_index = i

	return max_sum
```

**Comment by cmooney3** [comment]
Yeah that looks good to me.  It's tricky.  I thought it wouldn't work with an array of all negative numbers, but I think it does now that I'm looking at it closer.  Nice

**Comment by dabaitu** [comment]
Nice
